protostar/protostar_toml/protostar_toml_version_checker.py

```diff
@@ -0,0 +1,48 @@
+from typing import Any
+
+from protostar.protostar_exception import ProtostarException
+from protostar.protostar_toml.io.protostar_toml_reader import ProtostarTOMLReader
+from protostar.utils import VersionManager
+
+
+class ProtostarTOMLVersionChecker:
+    SKIP_FOR = ["init", "upgrade"]
+
+    def __init__(
+        self,
+        protostar_toml_reader: ProtostarTOMLReader,
+        version_manager: VersionManager,
+    ):
+        self._protostar_toml_reader = protostar_toml_reader
+        self._version_manager = version_manager
+
+    def run(self, args: Any):
+        if args.command in self.SKIP_FOR:
+            return
```

Comment:
`ProtostarTOMLVersionChecker` should be uncoupled from IO (CLI in this case).
I'd move this command to `ProtostarCLI`.

Reply:
You can modify `ProtostarCLI` to accept commands explicitly, e.g.

```python

def __init__(self, init_command: InitCommand, ...)
```
(This still breaks the open-close principle, however `ProtostarCLI` is more related to commands than `ProtostarTOMLVersionChecker`. To do it properly, the check should be in the command code. It would require a bit more code.)


Then you can constraint command name
```python
        if args.command in [self._init_command.name, self._upgrade_command.name]:
            # run checker
```